sample_apps/rust/src/utils/query_common.rs

```diff
@@ -0,0 +1,279 @@
+use aws_types::region::Region;
+use clap::Parser;
+use std::fs;
+use std::io::Write;
+
+const DEFAULT_DATABASE_NAME: &str = "devops_multi_sample_application";
+const DEFAULT_OUTPUT_FILE: &str = "query_results.log";
+const DEFAULT_REGION: &str = "us-east-1";
+const DEFAULT_TABLE_NAME: &str = "host_metrics_sample_application";
+
+#[derive(Parser, Debug)]
+#[command(version, about, long_about = None)]
+pub struct Args {
+    // The Timestream for LiveAnalytics database name to use for all queries
+    #[arg(short, long, default_value = DEFAULT_DATABASE_NAME)]
+    pub database_name: String,
+
+    // The full name of the output file to write query results to.
+    // For example, query_results.log
+    #[arg(short, long, default_value = DEFAULT_OUTPUT_FILE)]
+    pub output_file: String,
+
+    // The name of the AWS region to use for queries
+    #[arg(short, long, default_value = DEFAULT_REGION)]
+    pub region: String,
+
+    // The Timestream for LiveAnalytics table name to use for all queries
+    #[arg(short, long, default_value = DEFAULT_TABLE_NAME)]
+    pub table_name: String,
+}
+
+pub async fn get_connection(
+    region: &str,
+) -> Result<timestream_query::Client, timestream_query::Error> {
+    let config = aws_config::defaults(aws_config::BehaviorVersion::latest())
+        .region(Region::new(region.to_owned()))
+        .load()
+        .await;
+    let (client, reload) = timestream_query::Client::new(&config)
+        .with_endpoint_discovery_enabled()
+        .await
+        .expect("Failure to get the query client connection with Timestream");
+    tokio::task::spawn(reload.reload_task());
+    Ok(client)
+}
+
+pub fn write(mut file: &fs::File, s: String) -> Result<(), Error> {
+    let s_formatted = format!("{}\n", s);
+    file.write(s_formatted.as_bytes())
+        .map_err(|_| anyhow!("Failed to write to file"))?;
+    file.flush().map_err(|_| anyhow!("Failed to flush file"))?;
+    Ok(())
+}
+
+#[allow(dead_code)]
+pub fn process_scalar_type(data: &types::Datum) -> Result<String, Error> {
+    data.scalar_value
+        .clone()
+        .ok_or(anyhow!("Scalar value is None"))
+}
+
+#[allow(dead_code)]
+pub fn process_time_series_type(
+    data: &[types::TimeSeriesDataPoint],
+    column_info: &types::ColumnInfo,
+) -> Result<String, Error> {
+    let joined_string = data
+        .iter()
+        .map(|datum| -> Result<String, Error> {
+            let mut datum_string = String::new();
+            datum_string.push_str(&datum.time);
+            datum_string.push(':');
+
+            let column_type = column_info.r#type();
+            let column_type_ref = column_type.as_ref().ok_or(anyhow!("Column type is None"))?;
+            let scalar_type = column_type_ref.scalar_type.to_owned();
+            let scalar_type_ref = scalar_type.ok_or(anyhow!("Scalar type is None"));
+
+            let datum_value = datum.value.as_ref().ok_or(anyhow!("Datum value is None"))?;
+            let _scalar_value = datum_value
```

Comment:
Is this call necessary at all? Why are we ignoring the value.

Reply:
Not anymore. I have now removed this.